sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -92,7 +92,24 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
   /** Specifies how data is partitioned across different nodes in the cluster. */
   def outputPartitioning: Partitioning = UnknownPartitioning(0) // TODO: WRONG WIDTH!
 
-  /** Specifies any partition requirements on the input data for this operator. */
+  /**
+   * Specifies the data distribution requirements of all the children for this operator.
+   * By default it's [[UnspecifiedDistribution]], which means we don't have data distribution
+   * requirements for children.
+   *
+   * If an operator overwrites this method and provides specific distribution requirements for more
+   * than one child, it implies that these children's output should be co-partitioned, and these
+   * distribution requirements must be of same type, e.g., you can not require childA to be hash
+   * partitioned while requiring childB to be range partitioned, as they will never co-partitioned.
+   * If you require childA to be hash partitioned by column (a, b), and require childB to be hash
+   * partitioned by column (c, d), then the output of childA and childB should have the same number
+   * of partitions, and tuples of same value should be in the partitions of same index, e.g.,
+   * (a=1,b=2) and (c=1,d=2) are both in the second partition.
+   *
+   * [[BroadcastDistribution]] is a special case, because a broadcasted child can always be
+   * co-partitioned with others, so it's ok to mix [[BroadcastDistribution]] with other kind of
+   * distribution requirements.
```

Comment:
From `needCoPartitionChildIndex`, looks like `UnspecifiedDistribution` is also not counted as one of co-partition distributions?

Reply:
I mentioned this at the beginning of this doc: `provides specific distribution requirements for more than one child`, `specific distribution` excludes `UnspecifiedDistribution`, I'll make it more explicit.